daemon/appinfo.c

```diff
@@ -811,19 +824,31 @@ appinfo_parse_desktop(appinfo_t *self)
         g_free(tmp);
 
     /* Parse sailjail properties */
-    const gchar *group = SAILJAIL_SECTION_PRIMARY;
-    if( !g_key_file_has_group(ini, group) )
+    const gchar *group = NULL;
+    if( g_key_file_has_group(ini, SAILJAIL_SECTION_PRIMARY) )
+        group = SAILJAIL_SECTION_PRIMARY;
+    else if( g_key_file_has_group(ini, SAILJAIL_SECTION_SECONDARY) )
         group = SAILJAIL_SECTION_SECONDARY;
+    /* else: legacy app => use default profile */
 
-    tmp = keyfile_get_string(ini, group, SAILJAIL_KEY_ORGANIZATION_NAME, 0),
-        appinfo_set_organization_name(self, tmp),
-        g_free(tmp);
+    stringset_t *set;
+    if( group ) {
+        tmp = keyfile_get_string(ini, group, SAILJAIL_KEY_ORGANIZATION_NAME, 0),
+            appinfo_set_organization_name(self, tmp),
+            g_free(tmp);
 
-    tmp = keyfile_get_string(ini, group, SAILJAIL_KEY_APPLICATION_NAME, 0),
-        appinfo_set_application_name(self, tmp),
-        g_free(tmp);
+        tmp = keyfile_get_string(ini, group, SAILJAIL_KEY_APPLICATION_NAME, 0),
+            appinfo_set_application_name(self, tmp),
+            g_free(tmp);
 
-    stringset_t *set = keyfile_get_stringset(ini, group, SAILJAIL_KEY_PERMISSIONS);
+        set = keyfile_get_stringset(ini, group, SAILJAIL_KEY_PERMISSIONS);
+    }
+    else {
+        /* Read default profile from configuration */
+        set = config_stringset(appinfo_config(self),
```

Comment:
This is not apparently included via any header?

```
appinfo.c: In function ‘appinfo_parse_desktop’:
appinfo.c:848:15: warning: implicit declaration of function ‘config_stringset’; did you mean ‘keyfile_get_stringset’? [-Wimplicit-function-declaration]
         set = config_stringset(appinfo_config(self),
               ^~~~~~~~~~~~~~~~
               keyfile_get_stringset
appinfo.c:848:13: warning: assignment to ‘stringset_t *’ {aka ‘struct stringset_t *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
         set = config_stringset(appinfo_config(self),
             ^
^^^

Reply:
Added include, ran make depend and rebased.